src/components/Router/HistoryRouter.tsx

```diff
@@ -1,173 +1,223 @@
-      this.getStepType(this.state.step) === 'complete' ||
-      this.firstEnabledStep()
+      isNavigationDisabled ||
+      getInitialStep() ||
+      getStepType(state.step) === 'complete' ||
+      firstEnabledStep()
     )
   }
 
-  firstEnabledStep = () =>
-    this.getComponentsList()
-      .slice(0, this.state.step)
-      .every((c) => c.step.options?.skip)
-
-  initialStep = (): boolean =>
-    this.state.initialStep === this.state.step &&
-    this.state.flow === 'captureSteps'
-
-  changeFlowTo: ChangeFlowProp = (
+  const changeFlowTo: ChangeFlowProp = (
     newFlow,
     newStep = 0,
     excludeStepFromHistory = false
   ) => {
-    const { onFlowChange } = this.props
-    const { flow: previousFlow, step: previousUserStepIndex } = this.state
+    const { flow: previousFlow, step: previousUserStepIndex } = state
     if (previousFlow === newFlow) return
 
-    const previousUserStep = this.getComponentsList()[previousUserStepIndex]
+    const previousUserStep = getComponentsList(steps)[previousUserStepIndex]
 
     onFlowChange &&
-      onFlowChange(newFlow, newStep, previousFlow, {
-        userStepIndex: previousUserStepIndex,
-        clientStepIndex: previousUserStep.stepIndex,
-        clientStep: previousUserStep,
-      })
-    this.setStepIndex(newStep, newFlow, excludeStepFromHistory)
+      onFlowChange(
+        newFlow,
+        newStep,
+        previousFlow,
+        {
+          userStepIndex: previousUserStepIndex,
+          clientStepIndex: previousUserStep.stepIndex,
+          clientStep: previousUserStep,
+        },
+        steps
+      )
+    setStepIndex(newStep, newFlow, excludeStepFromHistory)
   }
 
-  nextStep = (): void => {
-    const { step: currentStep } = this.state
-    const componentsList = this.getComponentsList()
+  const nextStep = () => {
+    const { step: currentStep } = state
+    const componentsList = getComponentsList(steps)
 
     const nextStepComponent = componentsList
       .slice(currentStep + 1)
-      .find((c) => !c.step.options?.skip)
+      .find((c) => !c.step.skip)
 
-    if (!nextStepComponent) {
-      this.triggerOnComplete()
-    } else {
+    if (nextStepComponent) {
       const newStepIndex = componentsList.indexOf(nextStepComponent)
 
-      this.setStepIndex(newStepIndex)
+      setStepIndex(newStepIndex)
 
       const newStepType = nextStepComponent.step.type
-      const isNewStepType = this.props.currentStepType !== newStepType
+      const isNewStepType = currentStepType !== newStepType
       if (isNewStepType) {
-        this.props.actions.setCurrentStepType(newStepType)
+        actions.setCurrentStepType(newStepType)
       }
+    } else if (status !== 'finished') {
+      loadNextStep(() => {
+        setStepIndex(0, 'captureSteps')
+      })
     }
   }
 
-  triggerOnComplete = (): void => {
-    const { captures } = this.props
+  const triggerOnError: ErrorCallback = useCallback(
+    ({ response, status }) => {
+      if (status === 0) {
```

Comment:
For what do we check for `status === 0`?

Reply:
this is an existing code , only has been moved around.

Reply:
The Onfido API started returning 0 error codes, I don't remember for which cases.
Also, requests that fail due to lack of internet connection return a 0 zero status code.